src/core/server/saved_objects/import/regenerate_ids.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { v4 as uuidv4 } from 'uuid';
+import { SavedObject } from '../types';
+
+/**
+ * Takes an array of saved objects and returns an importIdMap of randomly-generated new IDs.
+ *
+ * @param objects The saved objects to generate new IDs for.
+ */
+export const regenerateIds = (objects: Array<SavedObject<{ title?: string }>>) => {
+  const importIdMap = objects.reduce((acc, object) => {
+    return acc.set(`${object.type}:${object.id}`, { id: uuidv4(), omitOriginId: true });
+  }, new Map<string, { id: string; omitOriginId?: boolean }>());
+  return { importIdMap };
```

Comment:
Originally it made sense to use named return values because I was implementing a "seed" that would deterministically regenerate IDs, but I scrapped that idea. I'll update this to simplify, thanks.